### ISSUE TITLE
Difficulties building on GNU/Linux (Arch Linux)

### ISSUE THREAD

**Issue by ahungry** [moebiusproject/moebiustoolkit#1]
Hi @suy - I've been trying to build and hitting various snags.

Firstly, I didn't see a mention of the need to run "qmake" (I just happened to see the hidden file in the repo root, and on a whim tried it - now the `make` command seems to work).

I have qt5 installed, and added the qt5-charts package, as it was missing from my qt5 dev files.

I got a little further in the build, but encountered this issue https://bugreports.qt.io/browse/QTBUG-93810 - fortunately the patch got me past it (see the attached file in the link's ticket).

Next, I run into a problem with: 

```
damagecalculatorpage.cpp:58:10: fatal error: tomlplusplus/include/toml++/toml.h: No such file or directory
   58 | #include "tomlplusplus/include/toml++/toml.h"
```
I went ahead and installed "tomlplusplus" from pacman (Arch's package repository), however this file seems to be attempting to include a local file in src/tomlplusplus - unfortunately, my src/tomlplusplus directory is empty.

Is there a way to build the contents of this directory?  Or is it expected I would copy something in here that's outside of VCS?  (apologies if it's in the readme, it's late and i read it at the start of this effort).

After the workarounds are enumerated, I think a really neat update to this project would be inclusion of a Dockerfile that built this system for GNU/Linux within it - I have some docker run commands that allow leveraging the host machine X server to make a docker app appear native (instead of the typical headless docker image) - it might be a nice packaging option for the moebius toolkit (as opposed to trying to build for many different *nixes).

**Comment by ahungry** [comment]
Oh, and I did try updating the tomlplusplus include to be the system level one - that just produced a variety of other errors, so I gave up on that path forward.

**Comment by suy** [comment]
Hello, and thanks for reporting!

First, yes, the app is built with qmake. I did not expect many people to build the application (glad that you did though!), so I did not consider documenting this. Also, due to my own bias, this was obvious to me, as the ".pro" files are exactly that, qmake files. I'll document it now, and the whole process.

> Is there a way to build the contents of this directory? Or is it expected I would copy something in here that's outside of VCS? (apologies if it's in the readme, it's late and i read it at the start of this effort).

The TOML library is inside the repo, but as a submodule. The first time you clone, if you pass `--recursive` to `git clone`, it does it for you. Now that you've already cloned, what you need to do is `git submodule update --init --recursive`. The `--recursive` in this case is highly optional, as it just recurses into the `tomlplusplus` directory to also do the same creation of the submodules of the TOML library, but I don't think we use those at all. I'm just saying that command out of habit and usual good practice.

I'll look further at the warning later. I guess you have a newer compiler version than me. Which version it is, BTW? If it stops people from compiling, I should reconsider my quite aggressive flags.

About Docker, I'll give it a thought, but I personally think Docker is the completely wrong option for this kind of things (building desktop applications). It often means that the application is so poorly structured that it builds in just a single OS setup with some specific setup. It should not be like that. The app should run and compile on a wide enough list of systems. I did not put much effort into this because I wasn't expecting much users. I've seen that I have gotten much more downloads of the releases that I would have expected.

If the application builds on many systems, though, adding a Dockerfile is gonna be fairly trivial, and I surely can do if anyone things is helpful for their way of doing.

Out of curiosity, did you try to use the AppImage? I know it will not work on older systems, but on Arch it should be the other way around, so I don't know why it could not work.

Thank you very much for the report. Let me know if there is something else that I can do.

**Comment by ahungry** [comment]
Thanks @suy !  I've used submodules a few times before, but it's never become habit to add the recursive flag to my standard git clone.

I haven't come across the AppImage format before - if this is the correct site for it https://appimage.org/, I have to say, that looks like a great ditsro method (I chalked it up to some alternate OSX install format other than the .dmg file :smile:) 

For versions, I'm not sure how to get more specific info than "qt5" (I see arch has some qt6 available as well).  My GCC/G++ is 12.2.0.

I'll update with submodule results in a bit
